### PR TITLE
added ruff pre-commit hook and removed flake8 and pylint

### PR DESCRIPTION
**Pull request by jstucke**
pylint has become so slow on modules like sqlalchemy that it is almost unusable. "ruff" is a linter written in Rust that can replace many checks of popular python linters and is lightning fast. The downside: it does not implement *all* checks that pylint does. On the upside it includes more checks from other linters we did not previously use like bugbear. Other not included  pylint checks 
are covered by included flake8 extensions. On the whole, it extends the linting capabilities and adds some new interesting checks.

- added ruff pre-commit hook (incl. auto-fix)
- removed pylint, isort and flake8 hooks (mostly replaced by ruff)
- updated version of black hook (to address warning)
- added toml check hook
- updated python version (to reflect removal of python 3.7 support #1024 )
- added ruff configuration to pyproject.toml
- the pylint configuration is still present in case someone want to manually run pylint

### CODE REVIEW

**Comment by maringuu** [comment]
We have a lot of `# pytlint:disable=...` comments in our code. Is there an equivalent to in ruff? Does ruff respect the existing comments?

Also ruff can replace isort too, so I thing we should to that.

**Comment by jstucke** [comment]
> Is there an equivalent to in ruff?

Yes: `# noqa: UP006` https://beta.ruff.rs/docs/tutorial/#ignoring-errors

> Also ruff can replace isort too, so I thing we should to that.

ruff also has a `--fix` option. I will look into how it is used together with pre-commit and how to configure the isort features in ruff.

**Comment by maringuu** [comment]
We can also replace our flake8 action by the ruff action. I'm not sure if we want that but I guess so.
https://beta.ruff.rs/docs/usage/#github-action

**Comment by jstucke** [comment]
> We can also replace our flake8 action by the ruff action. I'm not sure if we want that but I guess so. https://beta.ruff.rs/docs/usage/#github-action

The flake8 action should already be replaced (I replaced pylint, isort and flake8)

**Comment by maringuu** [comment]
> The flake8 action should already be replaced

I meant action as in github action. See https://github.com/fkie-cad/FACT_core/blob/ruff-linter/.github/workflows/lint_action.yml

File: pyproject.toml [hunk 1]
```diff
@@ -52,3 +44,33 @@ markers = [
     "cfg_defaults: Overwrite defaults for the testing config",
     "WebInterfaceUnitTestConfig: Configure the web_interface fixture",
 ]
+
+[tool.ruff]
+select = ["F", "E", "W", "C90", "N", "UP", "B", "A", "PL", "C4", "EXE", "ISC", "PIE", "T20", "PT", "RET", "SIM", "TCH",
+    "PTH", "PLR", "PLW", "ARG", "ERA", "RUF"]
+ignore = ["A003", "RUF001", "RUF002", "RUF003"]
+# Allow autofix for all enabled rules (when `--fix`) is provided.
```
Comment (maringuu) [inline pyproject.toml:52]: Not true, copy paste artifact?

**Review by maringuu — CHANGES_REQUESTED** [review]

**Review by maringuu — APPROVED** [review]
LGTM!

**Pull request merged**